### PR TITLE
Rootless with shmlocks was not working.

### PR DESCRIPTION
**Pull request by rhatdan**
This patch makes the path unigue to each UID.

Also cleans up some return code to return the path it is trying to lock.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
/approve
@mheon PTAL

**Comment by mheon** [comment]
LGTM
@giuseppe @baude @TomSweeneyRedHat @vrothberg PTAL

File: libpod/runtime.go [hunk 1]
```diff
@@ -705,12 +706,12 @@ func makeRuntime(runtime *Runtime) (err error) {
 
 		manager, err = lock.NewSHMLockManager(lockPath, runtime.config.NumLocks)
 		if err != nil {
-			return errors.Wrapf(err, "error creating SHM locks for libpod")
+			return err
 		}
 	} else {
 		manager, err = lock.OpenSHMLockManager(lockPath, runtime.config.NumLocks)
 		if err != nil {
-			return errors.Wrapf(err, "error opening libpod SHM locks")
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: Just curious why you removed the errors.Wrapf here and above.

**Comment by TomSweeneyRedHat** [comment]
Tests aren't hip.

File: libpod/runtime.go (hunk 1 above)
Comment (rhatdan) [reply libpod/runtime.go]: Underlying library has the same message now.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat  Tests pass now.

**Comment by baude** [comment]
LGTM, ill let @giuseppe give final lgtm

**Comment by giuseppe** [comment]
LGTM

**Pull request merged**